docs/libraries/libraries-with-cli.md

```diff
@@ -0,0 +1,660 @@
+
+* [Prerequisities](#prerequisites)
+* [How to target .NET Core](#how-to-target-net-core)
+* [How to target .NET Framework](#how-to-target-net-framework)
+* [How to target a Portable Class Library (PCL)](#how to target a portable class-library-pcl)
+* [How to cross-compile for .NET Core and .NET Framework](#how-to-cross-compile-for-net-core-and-net-framework)
+* [How to test libraries on .NET Core](#how-to-test-libraries-on-net-core)
+* [How to create a NuGet Package with your Library](#how-to-create-a-nuget-package-with-your-library)
+
+## Prerequisites
+
+You must have .NET Core installed on your machine.  This guide will focus on command line tooling because they work cross-platform and are what Visual Studio uses internally.  You have two options:
+
+* [DNX](http://aspnet.readthedocs.org/en/latest/getting-started/index.html), which has shipped with Visual Studio 2015 Update 1
+
+* [.NET CLI](http://dotnet.github.io/getting-started/), the tooling which will eventually replace DNX
+
+Either toolchain can be used to build and publish libraries today.
+
+The sections of this document dealing with older frameworks or Portable Class Libraries (PCLs) need the .NET Framework installed.  This is only supported on Windows.  
+
+The best way to install the .NET Framework is to [install Visual Studio](https://www.visualstudio.com/en-us/visual-studio-homepage-vs.aspx).  Alternatively, you can [install .NET Framework 4.6](https://www.microsoft.com/en-us/download/details.aspx?id=48130) as a standalone package.  This will allow you to target .NET Framework versions all the way down to .NET Framework 4.0.
+
+If you wish to support .NET Framework 2.0 or 3.5, you'll need to also [install .NET Framework 3.5](https://msdn.microsoft.com/en-us/library/hh506443(v=vs.110).  This will allow you to target .NET Framework versions all the way down to .NET Framework 2.0.
+
+## How to target .NET Core
+
+For the purposes of building a library, targeting ".NET Core" means targeting the .NET Standard Platform.  To see what that means exactly, see [.NET Platform Standard](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/standard-platform.md).
+
+Note that as of this time, `netstandardX.X` is not supported.  You'll have to use the `dotnet` monikers.  Here's a mapping:
+
+| dotnet | netstandard |
+|--------| ----------- |
+| dotnet51 | netstandard1.0 |
+| dotnet52 | netstandard1.1 |
+| dotnet53 | netstandard1.2 |
+| dotnet54 | netstandard1.3 |
+| dotnet55 | netstandard1.4 |
+
+
+A distilled version of what you need to know is this:
+
+You need to pick a version of `dotnetXX` (where `XX` is a version number) to add to your `project.json` file.
+
+1. If you don't have backwards compatibility requirements, target `dotnet55`.  Although this does support .NET Framework 4.6.x, it is the most modern target available.
```

Comment:
I still think that this is the wrong framing.

Proposed:

1. You can target the latest version - `dotnet55` - of the .NET Platform Standard, which will give you access to the most APIs. `dotnet55` is supported by .NET Framework 4.6 (and higher) and .NET Core 1.
2. You can target an earlier version of the .NET Platform Standard to target earlier .NET platforms and versions. Earlier .NET Platform Standard versions expose fewer APIs.  Example targets:

- Foo
- Bar
- Baz

[Refer to the platform mapping table](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/standard-platform.md#mapping-the-net-platform-standard-to-platforms) to choose the `dotnetXX` moniker you need to target.